sdk/storage/Azure.Storage.DataMovement/tests/StartTransferUploadDirectoryTests.cs

```diff
@@ -26,6 +26,20 @@ public StartTransferUploadDirectoryTests(bool async, BlobClientOptions.ServiceVe
             : base(async, serviceVersion, null /* RecordedTestMode.Record /* to re-record */)
         { }
 
+        private List<string> GetTestDirectoryTree(string parentDirectoryPath)
+        {
+            string subDirectory1 = CreateRandomDirectory(parentDirectoryPath);
+            string subDirectory2 = CreateRandomDirectory(parentDirectoryPath);
+
+            return new List<string>()
+            {
+                GetNewBlobName(),
+                GetNewBlobName(),
+                Path.Combine(subDirectory1, GetNewBlobName()),
+                Path.Combine(subDirectory2, GetNewBlobName()),
+            };
+        }
```

Comment:
Instead of creating the directories here and creating the files elsewhere, can directory creation logic be moved to `SetupDirectory()` where the starting files are created? I feel it makes more sense for setup of the directory to all happen at the same place.

I think this would revert your changes later in this file of the file list generation, as that code only changed to support this directory creation.

Reply:
Okay, I'm going to make a lot of changes `SetupDirectory()` then right now it only takes in the list of files, not the list of subdirectories as well. So instead of the paths, I'm going to take in the "Path" and check if it's a directory first.

Reply:
Done. I actually didn't have to change the setupdirectory at all, I just did the parsing for the parent directory and did the createifnotexists there.